tests/user_journey/test_rejection_path.py

```diff
@@ -0,0 +1,137 @@
+    a mentorship request until the mentee completes a task
+    - User B (mentee) sends a request to User A (mentor)
+    - User A (mentor) rejects request
+    """
+
+    def setUp(self):
+        super().setUp()
+
+        # Defining mentor for this test
+        self.mentor = UserModel(
+            name="Mentor A",
+            email=user1["email"],
+            username=user1["username"],
+            password=user1["password"],
+            terms_and_conditions_checked=user1["terms_and_conditions_checked"],
+        )
+
+        # Defining mentee for this test
+        self.mentee = UserModel(
+            name="Mentee B",
+            email=user2["email"],
+            username=user2["username"],
+            password=user2["password"],
+            terms_and_conditions_checked=user2["terms_and_conditions_checked"],
+        )
+
+        self.mentor.is_email_verified = True
+        self.mentor.available_to_mentor = True
+        self.mentee.is_email_verified = True
+        self.mentee.need_mentoring = True
+
+        db.session.add(self.mentor)
+        db.session.add(self.mentee)
+        db.session.commit()
+
+        self.test_description = "A request for mentorship"
+
+    def test_rejection_path(self):
+
+        mentor_auth_header = get_test_request_header(self.mentor.id)
+        mentee_auth_header = get_test_request_header(self.mentee.id)
+
+        # The mentee sends a new mentorship request to the mentor
+        request_body = {
+            "mentor_id": self.mentor.id,
+            "mentee_id": self.mentee.id,
+            "end_date": int((datetime.now() + timedelta(weeks=5)).timestamp()),
+            "notes": self.test_description,
+        }
+        send_request_response = self.client.post(
+            "/mentorship_relation/send_request",
+            headers=mentee_auth_header,
+            content_type="application/json",
+            data=json.dumps(request_body),
+        )
+
+        self.assertEqual(HTTPStatus.CREATED.value, send_request_response.status_code)
+        print("The mentorship request has been sent")
+
+        # Receiving the mentorship request on the mentor's side
+        received_requests_response = self.client.get(
+            "/mentorship_relations",
+            headers=mentor_auth_header,
+            content_type="application/json",
+        )
+        requests = json.loads(received_requests_response.data)
+
+        latest_mentorship_request = requests[0]
+        request_id = latest_mentorship_request["id"]
+        request_state = latest_mentorship_request["state"]
+
+        self.assertIsNotNone(request_id)
+        self.assertEqual(MentorshipRelationState.PENDING, request_state)
+
+        # The mentor rejects the request
+        reject_response = self.client.put(
+            f"/mentorship_relation/{request_id}/reject", headers=mentor_auth_header
+        )
+        self.assertEqual(HTTPStatus.OK.value, reject_response.status_code)
```

Comment:
Can you just remove the `.value` from the HTTPStatus enums please. It works either way in this tests situation so it should be fine without it ;) I can approve after.
```suggestion
        self.assertEqual(HTTPStatus.OK, reject_response.status_code)
```